### DISCUSSION TITLE
If the network cable is unpluged,  openthread clean thread network route/SRP information. Should it be OK from design?

### DISCUSSION THREAD

**Discussion by wallaceL** [openthread/openthread#7872] [Q&A]
If the network cable is unpluged,  openthread clean thread network route/SRP information. Should it be OK from the design?

**Comment by jwhui** [comment]
The OpenThread routing manager will handle state changes on the infrastructure interface: https://github.com/openthread/openthread/blob/dd631b2a1914b60223d527aafbca50b964cd9b7d/src/core/border_router/infra_if.cpp#L114

However, OpenThread does not currently clear the SRP server state when the border router role is disabled. This is intentional, since the device may choose to offer an SRP server even when there is no infrastructure link. This allows supporting DNS-SD in isolated Thread networks.

However, I could see some benefit to offering a mode that automatically disables SRP server when BR role is disabled.

**Reply by wallaceL** [reply]
understand, border routing manager is stopped. If I want to communicate with the device in thread mesh in this case, what should I do?

**Reply by jwhui** [reply]
Thread devices in the same mesh network can continue to communicate using mesh-local addresses.

**Reply by wallaceL** [reply]
mesh local address of the devices is not in SRP, and how do I get the mesh local address of device?

**Reply by jwhui** [reply]
SRP Client configuration allows specifying a list of IPv6 addresses or automatically picking the most appropriate address - see [src/cli/README_SRP_CLIENT.md](https://github.com/openthread/openthread/blob/main/src/cli/README_SRP_CLIENT.md#host-address) for more details.

**Comment by abtink** [comment]
> However, I could see some benefit to offering a mode that automatically disables SRP server when BR role is disabled.

I agree. I think this would be useful in some situations. 
I can later add a mechanism (new API or config) to *auto* enable/disable SRP server along BR (`RoutingManager`) functionality
